File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/di/module/application/ParserModule.java
package com.github.k1rakishou.chan.core.di.module.application;

import com.github.k1rakishou.chan.core.site.parser.search.SimpleCommentParser;
import com.github.k1rakishou.core_logger.Logger;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ParserModule {

    @Provides
    @Singleton
    public SimpleCommentParser provideChan4SimpleCommentParser() {
        Logger.deps("SimpleCommentParser");

        return new SimpleCommentParser();
    }

}

File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/di/module/application/ParserModule.kt
package com.github.k1rakishou.chan.core.di.module.application

import com.github.k1rakishou.chan.core.site.parser.search.SimpleCommentParser
import com.github.k1rakishou.core_logger.Logger.deps
import com.github.k1rakishou.v2.KurobaSettings
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ParserModule {
  @Provides
  @Singleton
  fun provideChan4SimpleCommentParser(kurobaSettings: KurobaSettings): SimpleCommentParser {
    deps("SimpleCommentParser")

    return SimpleCommentParser(kurobaSettings)
  }
}
